tests/product/test_collect.py

```diff
@@ -131,6 +139,28 @@ def test_basic_system_info(self, coordinator=None, hosts=None):
         self.assert_path_exists(self.cluster.master,
                                 OUTPUT_FILENAME_FOR_SYS_INFO)
 
+    def test_collect_system_info_dash_h_coord_worker(self):
+        self.setup_cluster(NoHadoopBareImageProvider(),
+                           self.STANDALONE_PRESTO_CLUSTER)
+        self.run_prestoadmin('server start')
+        actual = self.run_prestoadmin('collect system_info -H %(master)s,%(slave1)s')
+        self.test_basic_system_info(actual, self.cluster.internal_master, [self.cluster.master, self.cluster.slaves[0]])
+
+    def test_collect_system_info_dash_x_two_workers(self):
+        self.setup_cluster(NoHadoopBareImageProvider(),
+                           self.STANDALONE_PRESTO_CLUSTER)
+        self.run_prestoadmin('server start')
+        actual = self.run_prestoadmin('collect system_info -x %(slave2)s,%(slave3)s')
+        self.test_basic_system_info(actual, self.cluster.internal_master, [self.cluster.master, self.cluster.slaves[0]])
+
+    def test_collect_system_info_dash_h_invalid_host(self):
```

Comment:
Validating the command line arguments should be tested in a unit test; this shouldn't vary from command to command, and it *definitely* shouldn't rely on having a cluster up.